commcare_connect/opportunity/tasks.py

```diff
@@ -528,3 +534,27 @@ def send_invoice_paid_mail(opportunity_id, invoice_ids):
             recipient_list=recipient_emails,
             html_message=html_body,
         )
+
+
+@celery_app.task()
+def generate_automated_service_delivery_invoice():
+    for opportunity in Opportunity.objects.filter(active=True, managed=True):
+        start_date = get_start_date_for_invoice(opportunity)
+        end_date = get_end_date_for_invoice(start_date)
+        invoice_number = generate_invoice_number()
+        line_items = get_uninvoiced_visit_items(opportunity, start_date, end_date)
+        total_local_amount = sum(item["total_amount_local"] for item in line_items)
+        total_usd_amount = sum(item["total_amount_usd"] for item in line_items)
+
+        payment_invoice = PaymentInvoice(
+            opportunity=opportunity,
+            amount=total_local_amount,
+            amount_usd=total_usd_amount,
+            date=datetime.datetime.utcnow(),
+            start_date=start_date,
+            end_date=end_date,
+            status=InvoiceStatus.PENDING,
+            invoice_number=invoice_number,
+            service_delivery=True,
+        )
+        payment_invoice.save()
```

Comment:
What do you think about bulk saving? The queryset iterator supports chunking, so I'm thinking if we run over `n` elements we could populate a list in memory and bulk save it. Any thoughts?